### PR TITLE
CookieSession: expose HttpOnly and Secure attributes.

### PR DESCRIPTION
**Pull request by zedar**
#690 - `ClientSideSessionConfig` and `SessionIdCookieConfig` exposes `HttpOnly` and `Secure` attributes.
Defaults to: HttpOnly = true; Secure = false

### CODE REVIEW

**Comment by alkemist** [comment]
I wonder if we should try and somehow share these settings between the id and data cookies. Might be more complication than it's actually worth.

LGTM. +1 for merge.

**Comment by zedar** [comment]
I wonder if attributes like path, domain could be different between JSESSIONID and ratpack session cookies? Probably there is no reason. If not we can make `ClientSideSessionConfig` extension of `SessionIdCookieConfig`. What is your opinion?

**Comment by alkemist** [comment]
I think I'm leaning towards making them the same. In this case, we'd rename `SessionIdCookieConfig` to `SessionCookieConfig` and leave `ClientSideSessionConfig` with the same name. The client side session module would just depend on `SessionCookieConfig` for generic cookie related settings (e.g. httpsOnly), but keep the crypto settings for example.

**Comment by zedar** [comment]
Then `ClientSideSessionStore` will have a reference to 2 configs, `SessionCookieConfig` for basic and common attributes and `ClientSideSessionConfig` for more specific attributes. I think I can make such change today/tomorrow.

**Comment by alkemist** [comment]
ClientSideSessionStore will simply depend on `SessionCookieConfig`, like it essentially does. That is, the provider method for `ClientSideSessionStore` which just take it as a param.

**Comment by zedar** [comment]
ClientSideSessionStore depends on ClientSideSessionConfig and needs an access to common attributes (path, domain, httponly, secure) as well as client side specific (maxCookieSize, secureToken, ...). If we have 2 configs, one set by SessionModule and the other set be ClientSideSessionModule, then ClientSideSessionStore should have reference to both configs.

**Comment by zedar** [comment]
@alkemist could you please have a look at the last commit?

File: ratpack-session/src/main/java/ratpack/session/SessionCookieConfig.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  }
+
+  /**
+   * The name of the cookie for session id.
+   * <p>
+   * Defaults to: {@code JSESSIONID}
+   *
+   * @return the name of the cookie for session id
+   */
+  public String getName() {
```
Comment (alkemist) [inline ratpack-session/src/main/java/ratpack/session/SessionCookieConfig.java]: Need to rename this to make it clear it's for the session id cookie name.

`getIdName()` ?

**Comment by alkemist** [comment]
LGTM, besides SessionCookieConfig#getName()

**Comment by zedar** [comment]
`getIdName()` is much better name. Will change it and merge to master.

**Pull request merged**